### PR TITLE
Refactor to improve types for `selector-attribute-*` rules

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Part of #4496

> Is there anything in the PR that needs further explanation?

- Remove `// @ts-nocheck` to enable type-checking.
- Improve the code a bit for type safety.

### CODE REVIEW

File: lib/rules/selector-attribute-brackets-space-inside/index.js [hunk 1]
```diff
@@ -101,14 +100,18 @@ function rule(expectation, options, context) {
 				});
 			});
 
-			if (hasFixed) {
+			if (hasFixed && fixedSelector) {
```
Comment (ybiquitous) [inline lib/rules/selector-attribute-brackets-space-inside/index.js:103]: [note] `fixedSelector` is a `string` or `undefined`. See `lib/utils/parseSelector.js`.

File: lib/rules/selector-attribute-brackets-space-inside/index.js [hunk 2]
```diff
@@ -143,49 +149,53 @@ function rule(expectation, options, context) {
...
 			setAttrBefore(attrBefore.replace(/^\s*/, ''));
 		}
 	}
 
+	/**
+	 * @param {import('postcss-selector-parser').Attribute} attributeNode
+	 */
 	function fixAfter(attributeNode) {
-		let key;
+		const key = attributeNode.operator
```
Comment (ybiquitous) [inline lib/rules/selector-attribute-brackets-space-inside/index.js:163]: [note] Covert `let` to `const` for a more accurate variable type.

File: lib/rules/selector-attribute-brackets-space-inside/index.js [hunk 3]
```diff
@@ -143,49 +149,53 @@ function rule(expectation, options, context) {
...
 					},
 			  }
 			: {
-					after: (attributeNode.spaces[key] && attributeNode.spaces[key].after) || '',
+					after: (spaces && spaces.after) || '',
 					setAfter(fixed) {
 						if (!attributeNode.spaces[key]) attributeNode.spaces[key] = {};
 
+						// @ts-expect-error -- TS2532: Object is possibly 'undefined'.
 						attributeNode.spaces[key].after = fixed;
```
Comment (ybiquitous) [inline lib/rules/selector-attribute-brackets-space-inside/index.js:188]: [note] Unfortunately, the current TypeScript's compiler cannot detect that `attributeNode.spaces[key]` is present.

File: lib/rules/selector-attribute-operator-space-after/index.js [hunk 4]
```diff
@@ -34,12 +33,15 @@ function rule(expectation, options, context) {
...
+						/** @type {{ operatorAfter: string, setOperatorAfter: (fixed: string) => void }} */
 						const { operatorAfter, setOperatorAfter } = (() => {
 							const rawOperator = attributeNode.raws.operator;
 
 							if (rawOperator) {
 								return {
-									operatorAfter: rawOperator.slice(attributeNode.operator.length),
+									operatorAfter: rawOperator.slice(
+										attributeNode.operator ? attributeNode.operator.length : 0,
+									),
```
Comment (ybiquitous) [inline lib/rules/selector-attribute-operator-space-after/index.js:44]: [note] `str.slice(0) === str`

**Comment by ybiquitous** [comment]
I think a bit of coverage decrease is not a problem.

**Review by jeddy3 — APPROVED** [review]
Great, thank you!

**Pull request merged**